Sending a method to an Optional results in C error

**V version:** V 0.1.28 837df7c
**OS:** Linux

**What did you do?**
```v
import os

fn main() {
	os.ls('.').filter(it.ends_with('.v')) or { return }
}
```

**What did you expect to see?**
Some nice error.

**What did you see instead?**
```
/tmp/v/main.tmp.c: In function ‘main’:
/tmp/v/main.tmp.c:11456:37: error: ‘Option_array_string’ {aka ‘struct Option_array_string’} has no member named ‘len’
 int _t166_len = os__ls(tos_lit(".")).len;
                                     ^
/tmp/v/main.tmp.c:11457:30: error: invalid initializer
  Option_array_string _t166 = __new_array(0, _t166_len, sizeof(string));
                              ^~~~~~~~~~~
/tmp/v/main.tmp.c:11460:54: warning: passing argument 1 of ‘array_push’ from incompatible pointer type [-Wincompatible-pointer-types]
  if (string_ends_with(it, tos_lit(".v"))) array_push(&_t166, &it);
                                                      ^~~~~~
/tmp/v/main.tmp.c:4625:31: note: expected ‘array *’ {aka ‘struct array *’} but argument is of type ‘Option_array_string *’ {aka ‘struct Option_array_string *’}
 static void array_push(array* a, voidptr val) {
...
```
Might be related to https://github.com/vlang/v/issues/5453

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -735,6 +735,11 @@ pub fn (mut c Checker) call_method(mut call_expr ast.CallExpr) table.Type {
 	call_expr.left_type = left_type
 	left_type_sym := c.table.get_type_symbol(c.unwrap_generic(left_type))
 	method_name := call_expr.name
+
+	if left_type.has_flag(.optional) {
+		c.error('optional type cannot be called directly', call_expr.left.position())
+		return table.void_type
+	}
 	// TODO: remove this for actual methods, use only for compiler magic
 	// FIXME: Argument count != 1 will break these
 	if left_type_sym.kind == .array && method_name in ['filter', 'clone', 'repeat', 'reverse',

--- a/vlib/v/checker/tests/optional_type_call_err.out
+++ b/vlib/v/checker/tests/optional_type_call_err.out
@@ -1,0 +1,6 @@
+vlib/v/checker/tests/optional_type_call_err.v:4:5: error: optional type cannot be called directly
+    2 |
+    3 | fn main() {
+    4 |     os.ls('.').filter(it.ends_with('.v')) or { return }
+      |        ~~~~~~~
+    5 | }

--- a/vlib/v/checker/tests/optional_type_call_err.vv
+++ b/vlib/v/checker/tests/optional_type_call_err.vv
@@ -1,0 +1,5 @@
+import os
+
+fn main() {
+	os.ls('.').filter(it.ends_with('.v')) or { return }
+}

